### DISCUSSION TITLE
Testers needed for the new Sloeber version

### DISCUSSION THREAD

**Discussion by jantje** [Sloeber/arduino-eclipse-plugin#1660] [General]
I've been working over a year now on a new Sloeber version. Though it looks pretty much the same there have been a huge number of changes under the cover.
I can run unit tests for lots of functionality but there are holes in the unit tests and user interface is hard to test.
Therefore I would like experienced users to jump in and test the new Sloeber. 
I prefer experienced user who know what to expect from Sloeber.
Please reply below if you want to help out testing.

**Comment by ArminJo** [comment]
Hi Jaintje,
Thanks for giving us the Sloeber plugin 🥇 !!!
I would like to contribute and are willing to test the new version.
Best regards
Armin

**Comment by jantje** [comment]
An update.
I finally have sloeber working and issue #1126 solved.
That means testing can be done but I would to see more helping hands and #1625 fixed.
#1625 is important as I have missed many changes in the arduino framework so all the new stuff is likely to fail.

If you are not used to installing Sloeber from the actions feel free to get the "nightly" and start it to see how far you get. That is basically the first thing to test.

I propose to create issue with the name V5.0 testing
The main aim of the testing is to see everything still works from the gui perspective. Lots of stuff is tested with JUnit tests but that does not guarantee the GUI works.
Note that Version 5 is incompatible with the previous versions.
That means the configuration files created by Sloeber (and CDT in a way as I rewrote managedbuild that is part of CDT) are named different and/or have a different content.
The default behavior is also different.
Code is now (by default you can change it when creating a project) in the src folder and not in the root.
The build result is (by default you can change it in the project properties) in bin/${configName}
The default builder is the internal builder (no make required) but for some functionality (like burning the bootloader) make is still used. But that should be seamless from a user perspective.
Sloeber now adds the include directives automagically to the build commands.
At this point in time (I still need to find a way to do this better) the discovery will always open a command window.

Please join the testing and thanks for helping out.

**Reply by jantje** [reply]
Also changed
Better support for configuration changes.
The arduino folder now is arduino${configName}
That means that the libraries and cores remain available when switching configurations.

**Comment by ArminJo** [comment]
I istalled Sloever 4.4.3, ran check for updates, and restarted sloeber.
Then I added http://eclipse.baeyens.it/update/V4/nightly to Available Software sites checked it and ran check for updates again.
The only update I saw was Nebula Oscilloscope Widget	1.4.0.202406291220 , which throws the error:
```
An error occurred while collecting items to be installed
session context was:(profile=DefaultProfile, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Can't download artifact osgi.bundle,org.eclipse.nebula.widgets.oscilloscope,1.4.0.202406291220 required by org.eclipse.nebula.widgets.oscilloscope[1.4.0.202406291220], org.eclipse.jetty.logging[12.0.9], org.eclipse.nebula.widgets.oscilloscope.feature.feature.group[1.4.0.202406291220], org.eclipse.nebula.widgets.oscilloscope[1.4.0.202406291220] from any of the following repositories: 
file:/E:/Sloeber5/p2/org.eclipse.equinox.p2.core/cache/
file:/E:/Sloeber5/configuration/org.eclipse.osgi/550/data/listener_1925729951/
file:/E:/Sloeber5/.eclipseextension
file:/E:/Sloeber5/
https://download.eclipse.org/tools/cdt/releases/latest
https://download.eclipse.org/releases/latest/
https://download.eclipse.org/justj/jres/17/updates/release/latest/
https://download.eclipse.org/nebula/releases/latest/
https://eclipse.baeyens.it/update/V4/stable/
http://eclipse.baeyens.it/update/V4/nightly
https://download.eclipse.org/eclipse/updates/latest
Can't download artifact osgi.bundle,org.eclipse.nebula.widgets.oscilloscope.source,1.4.0.202406291220 required by org.eclipse.nebula.widgets.oscilloscope.source[1.4.0.202406291220], org.eclipse.nebula.widgets.oscilloscope.feature.feature.group[1.4.0.202406291220], org.eclipse.nebula.widgets.oscilloscope.source[1.4.0.202406291220] from any of the following repositories: 
file:/E:/Sloeber5/p2/org.eclipse.equinox.p2.core/cache/
file:/E:/Sloeber5/configuration/org.eclipse.osgi/550/data/listener_1925729951/
file:/E:/Sloeber5/.eclipseextension
file:/E:/Sloeber5/
https://download.eclipse.org/tools/cdt/releases/latest
https://download.eclipse.org/releases/latest/
https://download.eclipse.org/justj/jres/17/updates/release/latest/
https://download.eclipse.org/nebula/releases/latest/
https://eclipse.baeyens.it/update/V4/stable/
http://eclipse.baeyens.it/update/V4/nightly
https://download.eclipse.org/eclipse/updates/latest
Can't download artifact org.eclipse.update.feature,org.eclipse.nebula.widgets.oscilloscope.feature,1.4.0.202406291220 required by org.eclipse.nebula.widgets.oscilloscope.feature.feature.jar[1.4.0.202406291220], org.eclipse.nebula.widgets.oscilloscope.feature.feature.jar[1.4.0.202406291220], org.eclipse.nebula.widgets.oscilloscope.feature.feature.group[1.4.0.202406291220] from any of the following repositories: 
file:/E:/Sloeber5/p2/org.eclipse.equinox.p2.core/cache/
file:/E:/Sloeber5/configuration/org.eclipse.osgi/550/data/listener_1925729951/
file:/E:/Sloeber5/.eclipseextension
file:/E:/Sloeber5/
https://download.eclipse.org/tools/cdt/releases/latest
https://download.eclipse.org/releases/latest/
https://download.eclipse.org/justj/jres/17/updates/release/latest/
https://download.eclipse.org/nebula/releases/latest/
https://eclipse.baeyens.it/update/V4/stable/
http://eclipse.baeyens.it/update/V4/nightly
https://download.eclipse.org/eclipse/updates/latest
```
This is the installed version:
  Nebula Oscilloscope Widget	1.4.0.202402011804	org.eclipse.nebula.widgets.oscilloscope.feature.feature.group	Eclipse Nebula

And I saw no new sloeber packages / version.
This is the Version installed:
  sloeber	4.4.3.202402080117	io.sloeber.product	null

Do I have the 5.0 version now?

**Reply by jantje** [reply]
You do not have Sloeber 5.0.
You can not install sloeber 5.0 with the nightly.
I'll create a discussion on how to install "the nightly" and link it here once I've done that.
Note there are some build issues (weird things I can not reproduce locally) I hope to get fixed soon; until then all reported V5 issues remain unfixed in the nightly build.

**Reply by jantje** [reply]
Seems the build issues have magically disappear.
V5 is up and running again.

**Comment by jantje** [comment]
Here is how to install the "V5/nightly/actions" build https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1663

**Comment by jantje** [comment]
UPDATE:
I'm working on the migration code and I would like to get some projects to test.
Preferably "complicated projects" like with multiple configurations and compile options.
Note: only provide publicly available code
In your woirkspace rightclick the project you want to "donate" and select export->general->Archive file ....

**Reply by ArminJo** [reply]
An ESP32 project. 
https://www.dropbox.com/scl/fi/1ij16fxylshcn1btl2pcp/IRremote_ESP32.zip?rlkey=450ka1p3uqu90eyzb7ps1dw5m&dl=0

**Reply by jantje** [reply]
I tried to upgrade the project and the result did not build.
So I tired to import it into Sloeber V4 and after plenty of issues I came to the conclusion I hit the same problem in Sloeber V4.
This is the board and the error I get
![image](https://github.com/user-attachments/assets/3db3a76a-8e7d-4045-9efd-9dc1332fe58d)
The IR remote library I used is 4.4.0

FYI It seems to me this is a project to test the library (examples)? on multiple boards. In my book it is way easier to do so with a test script.

**Comment by ArminJo** [comment]
Sorry, this project was for testing the new 4.4.1 version, which is released now.
I use gihub actions for CI testing, but in case of error or ESP32 development, I require this Sloeber project.

**Reply by jantje** [reply]
No need for sorry
I'll try the new 4.4.1 version

**Comment by jantje** [comment]
Update: I've been distracted by stuff in the real world demanding my attention. Then I got ill and I'm still recovering. My brain functionality is still lower than usual making it hard to code.
Still the "convert to sloeber project" works (probably plenty of bugs but for simple projects it works)
Stuff I decided not to support is the "exclude from build" as it is to hard to identify the exclusions.
What should work:
1)the board is identified and used (if it exist in the new install)
2)Support for multiple configs

**Comment by jantje** [comment]
Another update.
Having had multiple problems with ESP32 and other weird things I think I finally think nailed it.
I also added support for libraries in the convert to Sloeber project.
It should be available to you if the latest action is successfully build.
Left to do: 
1)Fix reported bugs
2)update the visuals from V4 beryllium to V5 boron.

**Reply by TrippyLighting** [reply]
I am not really qualified enough to be a tester, but I downloaded and installed V5 on my 2017 Intel MacBook Pro anyway.
Do I just post an issue, or what's the best way to report "findings" for V5?

**Reply by jantje** [reply]
>I am not really qualified enough to be a tester

In my book you are very qualified.
I do lots of testing based on scripts but that does not represent the user experience.
For instance: Though all test are successful I noticed that when creating a project in a new Sloeber install the project was created wit "no file" instead of "ino file". This results in a bad user experience as default should work. (I still need to fix this)
As you have lots of experience with Sloeber and know "how it used to work" you can easily notice different behavior. Whenb the different behavior is not due to fixed stuff it is regression and a potential issue.
Also you have a mac :-)

>Do I just post an issue, or what's the best way to report "findings" for V5?

I propose to create issue with the name V5.0 testing

**Comment by JWindy** [comment]
Hey Jantje,

first and for all thank you so much for all your effort. I haven been using Sloeber about five years ago and was very happy to get proper IDE functionalty byond the Arduino IDE but without missing out on all the benefits of the Arduino framework. Now it's time for another project, which will run on an esp32 and which  moste likely will keep me bussy for a few months. I setup V4 a week ago and I'm impressed, how much has been improved and how little I had to configure. I might be slightly late, but I'll gladly give V5 a try and share found issues:D 

Since I haven't been working with Sloeber lately, it might make sense to run V5 as well as V4 in parallel to see if an issue is a bug or just a problem caused by the user :D You wrote above "that Version 5 is incompatible with the previous versions.". I assume this is for sure the config files, whiche are stored in .metadata, so I should create a new folder for V5. There seem to be also changes on the project files itself. E.g. source files in source folder. Can I work with V4 and V5 on the same project-folder? I assume the answer is no. Can I migrate the V4 projects to V5? Worst case would be to create another Workspace, copy the source file and use the "Convert to Sloeber project" function. If I have to go back to V4, I'll do this quick and dirty somewhere outside my tidy workspace:D

Johannes

**Comment by jantje** [comment]
First of all thanks for testing.
I think I fixed all reported issue and I would like to release V5.
Anyone thinks there are still open issues please point them out to me.

**Reply by ArminJo** [reply]
Hi Jaintje,
Thanks for your effort!

If I do Window > Perspective >reset the C/C++ Perspective in Beryllium I have this (customize Perspective menu entry)
![grafik](https://github.com/user-attachments/assets/947b421a-e3bd-4093-b118-cf18a5b99c13)
In Boron I have this
![grafik](https://github.com/user-attachments/assets/dadb0659-3e62-4896-8c6b-2a57fa91d49f)

Both have the same Eclipse base (latest updates all installed except Nebula widget).
And therefore configurations are not available in Boron 😢.

**Reply by jantje** [reply]
I have the following in boron in a fresh install
![image](https://github.com/user-attachments/assets/c2a6d820-a97e-4d58-8e13-81ce072a3096)

>And therefore configurations are not available in Boron

This is only the selection of availability of action sets in the gui for the default C/C++ perspective. IMHO "configurations are not available in Boron" is therefore not correct.
To me it looks like you should be able to simply activate the action set.

Can you check the oomph preferences?
![image](https://github.com/user-attachments/assets/ee99163e-054b-48b7-acd6-16ce95362893)
In the product oomph is 100% disabled/removed (so you can not find the preferences).

**Reply by ArminJo** [reply]
I have no oomph preferences available in my boron.
How do I make a fresh install? Just CDT and Repository.zip, or 4.4.3 and Repository.zip?

**Reply by ArminJo** [reply]
I fresh install (CDT + Repository.zip) fixed at least this problem.

**Reply by jantje** [reply]
A fresh install is taking the zip/tar (available in actions for testing or nightly) of your platform and uncompressing it.

>I have no oomph preferences available in my boron.

good.

**Comment by ArminJo** [comment]
If I open an existing 4.4 project the platform folder settings are lost, but still in .sproject.
Instead the last platform choosen on another project before the update is always selected.
If I set the platform and board, then I get 
```
**** Build of configuration _ReceiveDemo for project IRremote_Leonardo ****
Nothing to build for project IRremote_Leonardo
```
even after "clean project".

**Reply by jantje** [reply]
>If I open an existing 4.4 project

You need to do a convert to sloeber project

**Reply by ArminJo** [reply]
> You need to do a convert to sloeber project

How do I do it? manually?

**Reply by jantje** [reply]
yes. file->new->other->sloeber->convert to Sloeber project

**Reply by ArminJo** [reply]
After converting a multi configuration project, all 3 cpp are enabled. 
After **removing the old libraries directory manually** and disabling 2 of the cpp with "resouce Configuration" > "Exclude from Build" it worked.

And the old entry Config._ReceiveDemo.compile.sloeber.size.type=ARDUINO_WAY is "converted" to _ReceiveDemo.extension.sloeber.size.type=RAW_RESULT.

I know, programming the converting sucks and is tedious, but this also holds for me, doing it manually for all my projects. Its a dilemma.

**But at least, it works and starting ESP32 compilation is way faster than with 4.4.3** 😀 
T H A N K S

**Reply by jantje** [reply]
Converting the size command seemed hardert than one would expect.
Anyway the newest version "should do it".
Still testing needed.
Defaults to Arduino way.

**Comment by darek-i-rafal-i-zbychu** [comment]
I installed Sloeber 5 and plugins. Compilation works fine, but there is a problem with transferring files to the microcontroller.
The incorrect programmer is entered, because it should be "jtag2updi". The correct programmer is entered in the configuration. I have recently noticed that errors appear in the interpretation of data set by the user. Especially in the avrdude command line for sending data to the microcontroller.

Starting upload
Uploading project "Tiny816Blink" with "avrdude"


Launching: D:\Progel\Sloeber5_1\arduinoPlugin\packages\arduino\tools\avrdude\6.3.0-arduino17/bin/avrdude -CC:\Users\US3\Documents\Arduino\hardware\megaTinyCore2\megaavr/avrdude.conf -pattiny816 -cstk500v1 -PCOM5 -b19200 -Uflash:w:D:\Progel\Sloeber5_1\Projekty\Tiny816Blink\bin\Release/Tiny816Blink.hex:i 
Output:
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x03
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x03
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0x03
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0x03
avrdude: stk500_recv(): programmer is not responding

**Comment by PerennialNovice** [comment]
I'd like to download the latest and greatest version, but either I'm too dumb to find the download link on the Actions page, or the expired label means its no longer available for download...

**Comment by jantje** [comment]
@PerennialNovice 
I already moved V5 to master so the latest action build is the latest V5.
I haven't released it yet because the migration from V4 to V5 is not yet documented not fully completed (the library folder needs to be deleted)
Current latest build can be found here: https://github.com/Sloeber/arduino-eclipse-plugin/actions/runs/15801316789

**Comment by ghost** [comment]
following your link ... I cannot download the build:

<img width="520" height="430" alt="image" src="https://github.com/user-attachments/assets/c3549555-4711-42eb-80b5-71c6c556b83a" />

**Reply by jantje** [reply]
A build was just started.
Hopefully successfully (sometimes the build fails for a reason I do not understand and restarting it then mostly works)
https://github.com/Sloeber/arduino-eclipse-plugin/actions/runs/18079215479

**Reply by jantje** [reply]
Each time you havn't build a eclipse plugin for a couple of moths there will be build problems 🙄

**Reply by jantje** [reply]
So the build has been updated and was executed successfully here.
https://github.com/Sloeber/arduino-eclipse-plugin/actions/runs/18266874205

**Reply by PerennialNovice** [reply]
> So the build has been updated and was executed successfully here. https://github.com/Sloeber/arduino-eclipse-plugin/actions/runs/18266874205

Nince!
thank you very much

**Comment by jantje** [comment]
I noticed that yesterday as well.
Seems github deletes old builds (I thought it kept the latest build but I was wrong)
I didn't find a way to restart the build nor a way to keep the latest build so I decided to fix a issue and that will trigger a new build.
Unfortunately fixing the issue is not yet finished.
Hopefully in a couple of days I will get the issue fixed.
If you know a way to trigger the github build action in another way I'd be happy to learn.
Otherwise the only solutions I see are a local build (as described on the first page) or waiting a couple of days.

**Reply by ArminJo** [reply]
Hi Jaintje
if you add   `workflow_dispatch: # To run it manually`
you can trigger the build manually.

Example:
```
on:
  workflow_dispatch: # To run it manually
    description: 'manual build check'
  push: # see: https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request
    paths:
    - '**.ino'
    - '**.cpp'
    - ...
```

**Reply by jantje** [reply]
That seems to have done the trick.
Off course  checking the change in triggered the normal build action but I can see the button to start a manual build now.
